### PR TITLE
[service.autostop] 1.0.0

### PR DESCRIPTION
**Pull request by jbinkley60**
Description

Initial new Kodi Matrix addon service.autostop submission.

This is a simple Kodi service addon which monitors video and music playback. It allows users
to selectively stop paused or current playback after a certain amount of time.
Checklist:

[X ] My code follows the add-on rules and piracy stance of this project.
[X ] I have read the CONTRIBUTING document
[X ] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0
[X ] My code follows the add-on rules and piracy stance of this project.
[X ] I have read the CONTRIBUTING document
Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :

Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
add-on development wiki page.
Kodi pydocs provide information about the Python API
PEP8 codingstyle which is considered best practise but not mandatory.
This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at Codacy. You can create your own account as well to continuously monitor your python coding before submitting to repo.
Development questions can be asked in the add-on development section on the Kodi forum.

### CODE REVIEW

File: service.autostop/addon.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon  id="service.autostop" name="Autostop" version="1.0.0" provider-name="jeff@thebinks.com">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+  </requires>    
+  
+  <extension point="xbmc.service" library="service.py"/>
+
+  <extension point="xbmc.python.library" library="utilities.py"/>
```
Comment (basrieter) [inline service.autostop/addon.xml:9]: Why do you expose this one as a library? If you only use it internally, there is no need to do this.

**Review by basrieter — CHANGES_REQUESTED** [review]
Please review my comments.

File: service.autostop/common.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        return self.__printDebug(message, 3)
+
+    def debugplus(self, message):
+        return self.__printDebug(message, 3)
+
+    def __printDebug( self, msg, level=1 ):
+        if self.level >= level :
+            #msg=self.token_regex.sub("-Token=XXXXXXXXXX&", str(msg))
+            #msg=self.ip_regex.sub(".X.X.", msg)
+            print("%s%s -> %s : %s" % (self.main, self.sub, inspect.stack(0)[2][3], msg))
```
[the remark this answers is not in the document]
Comment (jbinkley60) [reply service.autostop/common.py]: This was old code from another addon.  the common.py file has been updated.

File: service.autostop/service.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    def onPlayBackStopped(self):
+        self.paflag = 0
+        self.plflag = 0
+
+             
+player = XBMCPlayer()
+ 
+monitor = xbmc.Monitor()      
+ 
+while True:
```
[the remark this answers is not in the document]
Comment (jbinkley60) [reply service.autostop/service.py]: This code in the service.py file has been updated as suggested.

File: service.autostop/service.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+pos = 0
+file = ''
+count = 0 
+pacount = 0
+plcount = 0
+
+xbmc.log("Autostop addon service started." , xbmc.LOGINFO)
+
+def settings(setting, value = None):
```
[the remark this answers is not in the document]
Comment (jbinkley60) [reply service.autostop/service.py]: This code in the service.py and utilities.py files have been updated as suggested.

File: service.autostop/addon.xml (hunk 1 above)
Comment (jbinkley60) [reply service.autostop/addon.xml:9]: This is called externally  typically via favourites.xml or keyboard.xml.  It is like a sleep button on an alarm clock.

**Review by basrieter — APPROVED** [review]

**Pull request merged**